### DISCUSSION TITLE
CPU/memory indicators not working when running Dockerised installation

### DISCUSSION THREAD

**Discussion by miohtama** [sagemathinc/cocalc#6713] [Q&A]
I am testing out Dockerirsed version of CoCalc (and ipykernel for Jupyter).

I noticed CPU and Memory indicators are permanently zero. I assume this is because I would need to grant Docker some special permission to read these values from the host system. (It could be also caused by some other fact, but this is my guess based on my prior experience working with Dockerised software.)

I hope to ask if  there exist any simple pointers or fixes available before I start to looking into this.

![image](https://github.com/sagemathinc/cocalc/assets/49922/a05da220-2b7f-4980-b84c-169d1e43c313)

**Comment by williamstein** [comment]
These definitely work in our Kubernetes version of cocalc.

@haraldschilly any thoughts?   You added these, and I don't know why they wouldn't work in cocalc-docker.   Maybe some command line program has to be installed, and it isn't?

**Comment by haraldschilly** [comment]
There is definitely no need to install any command line tool. All the info about processes is read directly from `/proc` or `/sys` files or wherever they are. Maybe there is a related problem, e.g. something with cgroups, and then it's not clear which processes belong to the project. … or the paths of these kernel system files are in a different location?

What's the value of the "kucalc" admin setting? Maybe this CPU/mem info should just not be shown if it is in the docker ("no" ?) setting.

Are there any relevant lines in the project's log? e.g. `project:project-info` or `project:ProjectStatusServer` ?

**Comment by miohtama** [comment]
Looks like Docker cannot fully read `/proc` by default according [to the discussion here](https://stackoverflow.com/questions/47072586/docker-access-host-proc). I will figure out a solution and post it here.

Any Docker process gets filtered down `/proc` and `/sys`.

**Comment by miohtama** [comment]
Hmm I just did `docker-compose exec cocalc /bin/bash` and then `htop` and looks like it is running within the container fine.

Where could I find the exact source code, package and method Cocalc is attempting to read the system information?

**Reply by haraldschilly** [reply]
One central point where it reads this is [in this file](https://github.com/sagemathinc/cocalc/blob/master/src/packages/project/project-info/server.ts). There could be just one little missing basic or a configuration is slightly off. I also don't know (without having to dig through all the code) how the cpu/mem indicator gets it's information. Maybe it's not even related to code in the file I linked to.
